lib/active_admin/namespace.rb

```diff
@@ -100,11 +109,25 @@ def root?
     #   Namespace.new(:root).module_name # => nil
     #
     def module_name
-      root? ? nil : @name.camelize
+      _name = @name
+      unless root?
+        if _name.is_a?(Array)
+          _name.map(&:to_s).map(&:camelize).join('::')
+        else
+          _name.camelize
+        end
+      end
     end
 
     def route_prefix
-      root? ? nil : @name
+      _name = @name
+      unless root?
+        if _name.is_a?(Array)
+          _name.map(&:to_s).map(&:to_s).join('_').underscore
```

Comment:
@siutin  not sure double map needed here , same propose  with one liner

Reply:
You are right, wrong copy-paste